plutus-pab/src/Cardano/Wallet/API.hs

```diff
@@ -5,16 +5,20 @@ module Cardano.Wallet.API
     ( API
     ) where
 
-import           Ledger            (PubKey, Value)
-import           Ledger.AddressMap (UtxoMap)
-import           Ledger.Slot       (Slot)
-import           Ledger.Tx         (Tx)
-import           Servant.API       (Get, JSON, NoContent, Post, ReqBody, (:<|>), (:>))
-import           Wallet.Effects    (Payment)
+import           Ledger                 (PubKey, Value)
+import           Ledger.AddressMap      (UtxoMap)
+import           Ledger.Slot            (Slot)
+import           Ledger.Tx              (Tx)
+import           Servant.API            (Capture, Get, JSON, NoContent, Post, ReqBody, (:<|>), (:>))
+import           Wallet.Effects         (Payment)
+import           Wallet.Emulator.Wallet (Wallet)
+
 
 type API
-    = "submit-txn" :> ReqBody '[JSON] Tx :> Post '[JSON] NoContent
-      :<|> "own-public-key" :> Get '[JSON] PubKey
-      :<|> "update-payment-with-change" :> ReqBody '[JSON] (Value, Payment) :> Post '[JSON] Payment
-      :<|> "wallet-slot" :> Get '[JSON] Slot
-      :<|> "own-outputs" :> Get '[JSON] UtxoMap
+    = "create" :> Post '[JSON] Wallet
+      :<|> Capture "walletId" Wallet :> "submit-txn" :> ReqBody '[JSON] Tx :> Post '[JSON] NoContent
```

Comment:
You can pull the `walletId` parameter out of the list:

```haskell
:<|> Capture "walletId" Wallet :> (
  "submit-txn" :> ReqBody '[JSON] Tx :> Post '[JSON] NoContent
  :<|> "own-public-key" :> Get '[JSON] PubKey
  ...
)
```

Reply:
This makes Client.hs/Server.hs code more complicated and less readable. I don't mind this duplication. If you don't mind I'd leave it as is.

Reply:
ok, fine with me.